### PR TITLE
getDevices manual test

### PR DESCRIPTION
**Pull request by gsinafirooz**
Add manual test for the newly implemented API getDevices

### CODE REVIEW

File: get_devices/web_app.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+const nordicUARTService = '6e400001-b5a3-f393-e0a9-e50e24dcca9e';
+// The test service defined in device_code.js
+const testService = '0b30acec-193e-11eb-adc1-0242ac120002';
+// The test characteristic defined in device_code.js
+const testCharacteristic = '0b30afd0-193e-11eb-adc1-0242ac120002';
+
+let gattServer = undefined;
+let btDeviceId = null;
+const urlParam = 'bt_device_id=';
+const btDeviceIdUrlParam = window.location.search.substring(window.location.search.indexOf(urlParam) + urlParam.length).split('&')[0];
```
Comment (reillyeon) [inline get_devices/web_app.js]: Use the [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) interface to parse and query the URL parameters.

File: get_devices/web_app.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+  if (gattServer) {
+    logInfo('Disconnecting from GATT.');
+    gattServer.disconnect();
+  }
+}
+
+async function refreshPage() {
+  logInfo('Refreshing the page');
+  window.location.replace(`${window.location.href.split("?")[0]}?bt_device_id=${btDeviceId}`);
```
Comment (reillyeon) [inline get_devices/web_app.js]: You can use [`URLSearchParams.toString()`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/toString) to build a new URL.

File: get_devices/web_app.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  var passed = false;
+  for (let i = 0; (i < pairedDevices.length) && !passed; i++) {
+    logInfo(`Bluetooth Device with name "${pairedDevices[i].name}" and id "${pairedDevices[i].id}" was detected.`);
+    passed = pairedDevices[i].id == btDeviceIdUrlParam;
+    gatt = pairedDevices[i].gatt;
+  }
+  if (passed) {
+    try {
+      logInfo(`Connecting to GATT server for device id "${btDeviceIdUrlParam}"...`);
+      gattServer = await gatt.connect();
```
Comment (reillyeon) [inline get_devices/web_app.js:112]: Try to get the primary service again as well.

File: get_devices/web_app.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  $('btn_start_pairing').disabled = false;
+  $('btn_refresh_page').disabled = false;
+  $('btn_check_permission').disabled = false;
+}
+
+async function init() {
+  if (!isBluetoothSupported()) {
+    console.log('Bluetooth not supported.');
+    $('bluetooth_available').style.display = 'none';
+    if (window.isSecureContext == 'https') {
```
Comment (reillyeon) [inline get_devices/web_app.js]: Fix this as you did in other tests.

**Review by reillyeon — CHANGES_REQUESTED** [review]

File: get_devices/web_app.js (hunk 1 above)
Comment (gsinafirooz) [reply get_devices/web_app.js]: [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) doesn't like `+` in URL parameters. I can either leave it as is or convert the ID to hex before passing it to the URL parameter.

File: get_devices/web_app.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  if (gattServer) {
+    logInfo('Disconnecting from GATT.');
+    gattServer.disconnect();
+  }
+}
+
+async function refreshPage() {
+  logInfo('Refreshing the page');
+  let params = new URLSearchParams();
+  params.set(urlParam, "MhY+FqGfRK7YbVNBsHJHaQ==");
```
Comment (reillyeon) [inline get_devices/web_app.js]: Set this to the device ID of the device that was selected earlier.

File: get_devices/web_app.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+      logInfo(`Connected to GATT, requesting service: ${testService}...`);
+      const service = await gattServer.getPrimaryService(testService);
+      assertEquals(service.device, device, 'service device mismatch');
+
+      logInfo(`Connected to service uuid:${service.uuid}, primary:${service.isPrimary}`);
+      logInfo(`Requesting characteristic ${testCharacteristic}...`);
+      const characteristic = await service.getCharacteristic(testCharacteristic);
+      assertEquals(characteristic.service, service, 'characteristic service mismatch');
+    }catch (error) {
```
Comment (reillyeon) [inline get_devices/web_app.js]:
```suggestion
    } catch (error) {
```

**Review by reillyeon — CHANGES_REQUESTED** [review]

File: get_devices/web_app.js (hunk 6 above)
Comment (reillyeon) [reply get_devices/web_app.js]: It looks like your most recent commit undid this fix.

**Review by reillyeon — APPROVED** [review]

**Pull request merged**